lib/network_utils.ml

```diff
@@ -905,6 +905,27 @@ module Proc = struct
 end
 
 module Ovs = struct
+  let match_multiple patterns s =
+    let rec loop = function
+    | [] -> None
+    | pattern :: rest ->
+      match Re.exec_opt pattern s with
+      | Some groups -> Some groups
+      | None -> loop rest
+    in
+    loop patterns
+
+  let error_handler script args stdout stderr exn =
+    let patterns = List.map Re.Perl.compile_pat [
```

Comment:
This compiles the patterns for every invocation of `error_handler` unless the compiler moves this code out. I think it is safer to have this at the top level of the enclosing module.